python/lsst/meas/algorithms/spatialModelPsf.cc

```diff
@@ -0,0 +1,72 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ * See the COPYRIGHT file
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "lsst/meas/algorithms/SpatialModelPsf.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace algorithms {
+
+namespace {
+template <typename PixelT>
+void declareFunctions(py::module &mod) {
+    using MI =
+        lsst::afw::image::MaskedImage<PixelT, lsst::afw::image::MaskPixel, lsst::afw::image::VariancePixel>;
+
+    mod.def("createKernelFromPsfCandidates", createKernelFromPsfCandidates<PixelT>, "psfCells"_a, "dims"_a,
+            "xy0"_a, "nEigenComponents"_a, "spatialOrder"_a, "ksize"_a, "nStarPerCell"_a = -1,
+            "constantWeight"_a = true, "border"_a = 3);
+    mod.def("countPsfCandidates", countPsfCandidates<PixelT>, "psfCells"_a, "nStarPerCell"_a = -1);
+    mod.def("fitSpatialKernelFromPsfCandidates",
+            (std::pair<bool, double>(*)(lsst::afw::math::Kernel *, lsst::afw::math::SpatialCellSet const &,
+                                        int const, double const,
+                                        double const))fitSpatialKernelFromPsfCandidates<PixelT>,
```

Comment:
I suggest matching the C++ code and using `ImageT`, then provide template parameter `afw::image::MaskedImage<float>`. That would make it a bit easier to follow the mapping to the original code and support adding wrappings for `afw::image::Image<T>` in the future (which the C++ supports but your wrapping does not). On the other hand, maybe we'll never need them.

Barring that, I suggest spelling out `MI` as `MaskedImageT`.